File: ThreadTest.java
class Worker extends Thread {
    public void run() {
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

public class ThreadTest {
    public static void main(String[] args) throws InterruptedException {
        int N = 9000;
        Worker[] workers = new Worker[N];

        long start = System.currentTimeMillis();

        for (int i = 0; i < N; i++) {
            workers[i] = new Worker();
            workers[i].start();
        }

        for (int i = 0; i < N; i++) {
            workers[i].join();
        }

        long end = System.currentTimeMillis();
        System.out.printf("Java: Took %.2f seconds%n", (end - start) / 1000.0);
    }
}

File: thread_test.rs
use std::thread;
use std::time::{Duration, Instant};

fn main() {
    let n = 9000;
    let start = Instant::now();

    let mut handles = vec![];

    for _ in 0..n {
        let handle = thread::spawn(|| {
            thread::sleep(Duration::from_secs(2));
        });
        handles.push(handle);
    }

    for handle in handles {
        handle.join().unwrap();
    }

    let elapsed = start.elapsed();
    println!("Rust: Took {:.2} seconds", elapsed.as_secs_f64());
}

